<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CameraWindow</class>
 <widget class="QMainWindow" name="CameraWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CameraWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* 整体背景和字体 */
QWidget {
    background-color: #e8f4fd;
    font-family: &quot;Segoe UI&quot;, &quot;Microsoft YaHei&quot;;
    font-size: 14px;
    color: #2d3436;
}

/* 下拉框样式 */
QComboBox {
    background-color: #ffffff;
    border: 1px solid #74b9ff;
    border-radius: 6px;
    padding: 4px 8px;
    min-height: 30px;
}

QComboBox:hover {
    border: 1px solid #0984e3;
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border-left: 1px solid #dfe6e9;
    background: #d0e6fb;
}

QComboBox QAbstractItemView {
    background: #ffffff;
    border: 1px solid #74b9ff;
    selection-background-color: #74b9ff;
    selection-color: #ffffff;
}

/* 按钮样式 */
QPushButton {
    background-color: #d0e6fb;
    border: 1px solid #74b9ff;
    border-radius: 8px;
    padding: 6px 12px;
    min-height: 32px;
}

QPushButton:hover {
    background-color: #a2d2ff;
    border: 1px solid #0984e3;
}

QPushButton:pressed {
    background-color: #74b9ff;
    color: white;
}

/* 视频显示窗口边框 */
QVideoWidget {
    border: 2px solid #74b9ff;
    border-radius: 6px;
    background-color: black;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="3,1">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="CameraDisplayWidget" name="camera_display_widget" native="true"/>
    </item>
    <item>
     <widget class="QWidget" name="widget_2" native="true">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QWidget" name="widget_3" native="true">
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <widget class="QWidget" name="save_widget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,1">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Image saved At:</string>
               </property>
               <property name="wordWrap">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEdit">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_redirect">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>46</height>
             </size>
            </property>
            <property name="text">
             <string>redirect</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QWidget" name="camera_selector_widget" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_3">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Camera:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBox">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>40</height>
                </size>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_oppose_camera">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>46</height>
             </size>
            </property>
            <property name="text">
             <string>Open</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_capture_once">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>46</height>
             </size>
            </property>
            <property name="text">
             <string>Photo!</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_refresh">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>46</height>
             </size>
            </property>
            <property name="text">
             <string>Refresh</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_exit">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>46</height>
             </size>
            </property>
            <property name="text">
             <string>Exit</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonStyle::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionfresh"/>
   <addaction name="actionphoto"/>
   <addaction name="actionexit"/>
  </widget>
  <action name="actionexit">
   <property name="icon">
    <iconset resource="../sources/icons.qrc">
     <normaloff>:/icons/exit.png</normaloff>:/icons/exit.png</iconset>
   </property>
   <property name="text">
    <string>exit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionfresh">
   <property name="icon">
    <iconset resource="../sources/icons.qrc">
     <normaloff>:/icons/refresh.png</normaloff>:/icons/refresh.png</iconset>
   </property>
   <property name="text">
    <string>fresh</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
  <action name="actionphoto">
   <property name="icon">
    <iconset resource="../sources/icons.qrc">
     <normaloff>:/icons/capture.png</normaloff>:/icons/capture.png</iconset>
   </property>
   <property name="text">
    <string>photo</string>
   </property>
   <property name="menuRole">
    <enum>QAction::MenuRole::NoRole</enum>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CameraDisplayWidget</class>
   <extends>QWidget</extends>
   <header>CameraDisplayWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../sources/icons.qrc"/>
 </resources>
 <connections/>
</ui>
